apps/studio/src/pageComponents/main/JSCodeshiftRender.tsx

```diff
@@ -29,7 +31,15 @@ const MonacoDiffEditor = dynamic(
 
 export const useCodeDiff = () => {
 	const { setEvents, events } = useLogStore();
-	const { engine, inputSnippet, afterInputRanges } = useSnippetStore();
+	const { engine, afterInputRanges } = useSnippetStore();
+
+	const { selectFirst } = useFilesStore();
+
+	const inputSnippet =
+		selectFirst({
+			parent: DEFAULT_TEST_FIXTURE_DIR.hashDigest,
```

Comment:
maybe we can also add something like 'rootNode' to know exactly to which group a given file belongs?

Reply:
or even something like origin which will help us to build stronger type check. In for all files origin snippet names can be only before/after/output what will prevent from typeos